langstream-agents/langstream-agent-http-request/src/main/java/ai/langstream/agents/queryhttp/HttpRequestAgent.java

```diff
@@ -0,0 +1,187 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.agents.queryhttp;
+
+import ai.langstream.ai.agents.commons.JsonRecord;
+import ai.langstream.ai.agents.commons.TransformContext;
+import ai.langstream.api.runner.code.AbstractAgentCode;
+import ai.langstream.api.runner.code.AgentContext;
+import ai.langstream.api.runner.code.AgentProcessor;
+import ai.langstream.api.runner.code.Record;
+import ai.langstream.api.runner.code.RecordSink;
+import ai.langstream.api.runtime.ComponentType;
+import ai.langstream.api.util.ConfigurationUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.samskivert.mustache.Mustache;
+import com.samskivert.mustache.Template;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.avro.Schema;
+
+@Slf4j
+public class HttpRequestAgent extends AbstractAgentCode implements AgentProcessor {
+
+    static final ObjectMapper mapper = new ObjectMapper();
+    private final Map<Schema, Schema> avroValueSchemaCache = new ConcurrentHashMap<>();
+
+    private final Map<Schema, Schema> avroKeySchemaCache = new ConcurrentHashMap<>();
+
+    private AgentContext agentContext;
+    private ExecutorService executor;
+    private HttpClient httpClient;
+    private Template urlTemplate;
+    private String method;
+    private Map<String, Template> headersTemplates;
+    private Template bodyTemplate;
+    private String outputFieldName;
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void init(Map<String, Object> configuration) {
+        final String url =
+                ConfigurationUtils.requiredNonEmptyField(
+                        configuration, "url", () -> "http-request agent");
+        this.outputFieldName =
+                ConfigurationUtils.requiredNonEmptyField(
+                        configuration, "output-field", () -> "http-request agent");
+        this.method = ConfigurationUtils.getString("method", "GET", configuration);
+        final String body = ConfigurationUtils.getString("body", null, configuration);
```

Comment:
pleae make the user agent configurable, also the handling of redirections and cookies, like we do in the webcrawler